examples/src/chains/retrieval_qa_sources.ts

```diff
@@ -4,27 +4,74 @@ import { HNSWLib } from "langchain/vectorstores/hnswlib";
 import { OpenAIEmbeddings } from "langchain/embeddings/openai";
 import { RecursiveCharacterTextSplitter } from "langchain/text_splitter";
 import * as fs from "fs";
+import { PromptTemplate } from "langchain/prompts";
+import { StringOutputParser } from "langchain/schema/output_parser";
+import { Document } from "langchain/document";
 
-// Initialize the LLM to use to answer the question.
-const model = new OpenAI({});
-const text = fs.readFileSync("data/state_of_the_union_2022.txt", "utf8");
-const textSplitter = new RecursiveCharacterTextSplitter({ chunkSize: 1000 });
-const docs = await textSplitter.createDocuments([text]);
+const text = fs.readFileSync("examples/state_of_the_union.txt", "utf8");
 
-// Create a vector store from the documents.
-const vectorStore = await HNSWLib.fromDocuments(docs, new OpenAIEmbeddings());
+const serializeDocs = (docs: Array<Document>): string =>
+  docs.map((doc) => doc.pageContent).join("\n");
 
-// Create a chain that uses a map reduce chain and HNSWLib vector store.
-const chain = RetrievalQAChain.fromLLM(model, vectorStore.asRetriever(), {
-  returnSourceDocuments: true, // Can also be passed into the constructor
+const query = "What did the president say about Justice Breyer?";
+
+async function retrievalQAWithSources({
+  query,
+  text,
+}: {
+  /**
+   * The query to perform against the documents.
+   */
+  query: string;
+  /**
+   * The text to be chunked and used as context.
+   */
+  text: string;
+}): Promise<{ result: string; sourceDocuments: Array<Document> }> {
+  // Initialize the LLM to use to answer the question.
+  const model = new OpenAI({});
```

Comment:
Use `ChatOpenAI` for the later GPT-3.5 models or use `OpenAI` with `gpt-3.5-turbo-instruct`

Reply:
Refactored to `ChatOpenAI` and use `ChatPromptTemplate` w/ system & human prompt (did the same for other examples except `retrieval_qa_custom`)